packages/rrweb-snapshot/src/css.ts

```diff
@@ -437,15 +437,87 @@ export function parse(css: string, options: ParserOptions = {}) {
     }
     /* @fix Remove all comments from selectors
      * http://ostermiller.org/findcomment.html */
-    return trim(m[0])
+    const splitSelectors = trim(m[0])
       .replace(/\/\*([^*]|[\r\n]|(\*+([^*/]|[\r\n])))*\*\/+/g, '')
       .replace(/"(?:\\"|[^"])*"|'(?:\\'|[^'])*'/g, (m) => {
         return m.replace(/,/g, '\u200C');
       })
-      .split(/\s*(?![^(]*\)),\s*/)
-      .map((s) => {
+      .split(/\s*(?![^(]*\)),\s*/);
+
+    if (splitSelectors.length <= 1) {
+      return splitSelectors.map((s) => {
         return s.replace(/\u200C/g, ',');
       });
+    }
+
+    // For each selector, need to check if we properly split on `,`
+    // Example case where selector is:
+    // .bar:has(input:is(:disabled), button:is(:disabled))
+    let i = 0;
+    let j = 0;
+    const len = splitSelectors.length;
+    const finalSelectors = [];
+    while (i < len) {
```

Comment:
The logic is as follows:

* Iterate through the split selectors and look for the first selector with an opening parenthesis
* If opening paren is found, look through the rest of the selectors to look for a matching number of closing parentheses 
  * If closing paren is found, merge together all selectors between the opening and closing
    * Move iterator index to the next selector and continue
  * If no closing paren is found, assume that selectors were correctly split